### PR TITLE
ConfigDB Instrument State OR filter + Remove Integer Field

### PR DESCRIPTION
**Pull request by mgdaily**
Make Instrument state a CharField with a set of choices. Update the filter to use a MultiSelectFilter with those same choices. Add migrations to remove the integer field and automatically convert to the new char field.

I felt like it was probably best practice to keep these migrations in VCS rather than doing them by hand, so that's what I did. I tested reversing 34->32 locally and it worked great. 32->34 was done and deployed to http://configdb-dev.lco.gtn

http://configdb-dev.lco.gtn/instruments/?state=SCHEDULABLE
returns 8 instruments
http://configdb-dev.lco.gtn/instruments/?state=DISABLED
returns 37 instruments
to OR the two states:
http://configdb-dev.lco.gtn/instruments/?state=DISABLED&state=SCHEDULABLE
(returns 45)

### CODE REVIEW

File: configdb/hardware/tests.py [hunk 1]
```diff
@@ -61,13 +62,23 @@ def test_write_instrument(self):
...
 
     def test_patch_instrument(self):
-        self.assertEqual(self.instrument.state, Instrument.DISABLED)
+        self.assertEqual(self.instrument.state, 'DISABLED')
 
         self.client.login(username='tst_user', password='tst_pass')
         self.client.patch('/instruments/{}/'.format(self.instrument.pk), json.dumps({'state': 'MANUAL'}),
                           content_type='application/json')
         self.instrument.refresh_from_db()
-        self.assertEqual(self.instrument.state, Instrument.MANUAL)
```
Comment (mgdaily) [inline configdb/hardware/tests.py:70]: I kind of like having the `Instrument.MANUAL` enum-like states defined in the Instrument model. As I have it now, they're just gonna be strings but I am happy to make it work like an enum too. Opinions?

File: configdb/hardware/serializers.py [hunk 2]
```diff
@@ -6,22 +6,6 @@
     CameraType, GenericModeGroup, InstrumentType, ConfigurationTypeProperties
 )
 
-
-class StateField(serializers.IntegerField):
```
Comment (mgdaily) [inline configdb/hardware/serializers.py:10]: Not needed anymore, thankfully

File: configdb/hardware/migrations/0034_copy_instrument_state.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+from django.db import migrations
```
Comment (mgdaily) [inline configdb/hardware/migrations/0034_copy_instrument_state.py:1]: Decided to do this just to keep a complete historical record of the evolution of the DB.

File: configdb/hardware/migrations/0033_alter_instrument_state.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Generated by Django 4.0.3 on 2023-01-09 23:18
+
```
Comment (mgdaily) [inline configdb/hardware/migrations/0033_alter_instrument_state.py:2]: Migration part 1 just renames the integer state field to `state_int` and adds a new CharField `state`

**Review by markBowman — APPROVED** [review]
Works for me

File: configdb/hardware/tests.py (hunk 1 above)
Comment (jnation3406) [reply configdb/hardware/tests.py:70]: Either way is fine. I agree enums look a little nicer and prevent you from using magic strings in tests and stuff.

**Review by jnation3406 — APPROVED** [review]
Looks good, I like that the migration is reversible!

**Pull request merged**